specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -0,0 +1,5247 @@
+              }
+            },
+            "models": {
+              "description": "The list of models.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Model"
+              }
+            },
+            "environmentImageRequest": {
+              "description": "The Environment, models and assets used for inferencing.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/EnvironmentImageRequest"
+                }
+              ]
+            },
+            "vnetConfiguration": {
+              "description": "The virtual network configuration.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/VnetConfiguration"
+                }
+              ]
+            },
+            "encryptionProperties": {
+              "description": "The encryption properties.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/EncryptionProperties"
+                }
+              ]
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "ACI"
+    },
+    "ContainerResourceRequirements": {
+      "description": "The resource requirements for the container (cpu and memory).",
+      "type": "object",
+      "properties": {
+        "cpu": {
+          "format": "double",
+          "description": "The number of CPU cores on the container.",
+          "type": "number",
+          "example": 4
+        },
+        "memoryInGB": {
+          "format": "double",
+          "description": "The amount of memory on the container in GB.",
+          "type": "number",
+          "example": 64
+        },
+        "gpu": {
+          "format": "int32",
+          "description": "The number of GPU cores in the container.",
+          "type": "integer"
+        },
+        "fpga": {
+          "format": "int32",
+          "description": "The number of FPGA PCIE devices exposed to the container. Must be multiple of 2.",
+          "type": "integer"
+        }
+      }
+    },
+    "ModelDataCollection": {
+      "description": "The Model data collection properties.",
+      "type": "object",
+      "properties": {
+        "eventHubEnabled": {
+          "description": "Option for enabling/disabling Event Hub.",
+          "type": "boolean"
+        },
+        "storageEnabled": {
+          "description": "Option for enabling/disabling storage.",
+          "type": "boolean"
+        }
+      }
+    },
+    "VnetConfiguration": {
+      "type": "object",
+      "properties": {
+        "vnetName": {
+          "description": "The name of the virtual network.",
+          "type": "string"
+        },
+        "subnetName": {
+          "description": "The name of the virtual network subnet.",
+          "type": "string"
+        }
+      }
+    },
+    "EncryptionProperties": {
+      "type": "object",
+      "properties": {
+        "vaultBaseUrl": {
```

Comment:
why can't this take the full URL to the key (similar to BYOK https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#customer-managed-key-encryption)

Reply:
we just want to maintain parity with our existing [sdk/cli](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.webservice.aci.encryptionproperties?view=azure-ml-py), that accepts individual params for vaultUrl, keyName and keyVersion.